### ISSUE TITLE
Support async getQuery

### ISSUE THREAD

**Issue by taion** [relay-tools/found-relay#297]
Follow-up feature request from https://github.com/relay-tools/found-relay/issues/294

**Comment by jquense** [comment]
isn't this better solved by persistent queries?

**Comment by taion** [comment]
Probably.

**Comment by dminkovsky** [comment]
I don't know, I liked your reasoning. I think I was mislead by the apparent tidiness of keeping the query with the module. If this capability existed, it would have lead me down the wrong path?

**Comment by taion** [comment]
Yeah, we probably don't want to do anything here. It probably makes sense to use persistent queries and only send up the query hash, in cases where we want to omit the weight of queries from the main bundle.

**Comment by tomhalley** [comment]
Sorry for reviving this but, I'm currently working on a project where the cost of moving to persisted queries on the backend far outweighs the benefit of a reduced bundle size on the frontend, but if we had async getQuery, we'd still be able to get that benefit without the cost, so this would still be a useful thing to have.
